hashicorp/consul
# [NET-5688] APIGateway UI Topology Fixes

### Description

After the move to the native consul api gateways, the UI topology view around API gateways was broken in that API Gateways would just not show in the view. 

![image](https://github.com/hashicorp/consul/assets/16313206/25940dc2-7e2a-4b2b-8fe2-1e5abae03559)
![image](https://github.com/hashicorp/consul/assets/16313206/7ff7729e-650c-4bba-97f6-663047c13157)

### Testing & Reproduction steps

1. Compile a local version of consul without these changes using `make dev-docker` from consul
1. clone https://github.com/hashicorp/shared-consul-examples/ and run `make start` from the `k8s/basic-gw` directory
1. this will setup a port forward, in your browser visit `localhost:8501` and view the service/gateway topology views to see the missing parts
1. switch to this branch in the consul repo and re-run `make dev-docker`
1. use `ctrl-c` on the example where port-forwarding is running, then run `make reload`, this will reinstall consul with the new changes
1. go to `localhost:8051` again to see the topology views with the api gateway present
### Links



### PR Checklist
